### ISSUE TITLE
Getting, then later writing, a symbol's ID with v1.1 BinaryWriter?

### ISSUE THREAD

**Issue by barries** [amazon-ion/ion-rust#888]
In the telemetry framework I'm writing, it would be nice (tm) to be able to acquire symbol IDs for `&str`s, then emit as needed. My motivation is to be able to hold onto a symbol like the currently executing function name without having to do a heap allocation to hold that name between function entry and exit.

Is that a likely feature one day?

Two possible API concepts:

```rust
let symbol_id = ion_writer.write_symbol("foo")?;
... time passes ...
ion_writer.write(symbol_id)
```

```rust
let symbol_id = ion_writer.symbol_table().get("foo");
ion_writer.write(symbol_id)
... time passes ...
ion_writer.write(symbol_id)
```

**Comment by zslayton** [comment]
The latter is meant to be supported, but it looks like the `symbol_table()` method on the value writer isn't `pub`.

**Comment by zslayton** [comment]
This issue (along with #887) was fixed by PR #889.

**Comment by barries** [comment]
@zslayton, thanks for making SymbolTable public; given that `add_symbol_for_text()` is not `pub`, is there a way to add symbols directly (like the old `intern()` method, before writing them?

Or do I need to write one, then look it up using the same string (which is a bit of extra complexity on the call side and seems inefficient)?
